src/features/ajax/aggregate/index.js

```diff
@@ -19,15 +19,23 @@ export class Aggregate extends AggregateBase {
   static featureName = FEATURE_NAME
   constructor (agentIdentifier, aggregator) {
     super(agentIdentifier, aggregator, FEATURE_NAME)
+    const agentInit = getConfiguration(agentIdentifier)
+    const allAjaxIsEnabled = agentInit.ajax.enabled !== false
+
+    register('xhr', storeXhr, this.featureName, this.ee)
+    if (!allAjaxIsEnabled) return // feature will only collect timeslice metrics & ajax trace nodes if it's not fully enabled
```

Comment:
Was this change for session trace or replay?

Reply:
Should the return be after the denyList set?

Reply:
Need to add test that disabled ajax still harvest metrics & ajax trace nodes for sure

Reply:
Oh so race between deny list being set and `storeXhr` doesn't matter because storeXhr would still early return on init config (not subject to race) and not execute any code dependent on deny list.

Reply:
Is this going to break the drain behavior since this can return before invoking drain()

Reply:
^Oh yeah, thanks forgot about that drain line at the bottom. Yeah need to drain before early exit fs.

Reply:
Verified that if ajax enabled config is false, it won't be imported at all (since at least v1220) hence no need to test for timeslice & ajax nodes when it's disabled.